src/cirrus/lib/process_payload.py

```diff
@@ -198,55 +219,90 @@ def __call__(self, wfem) -> str | None:
         if not payload_bucket:
             raise ValueError("env var CIRRUS_PAYLOAD_BUCKET must be defined")
 
-        arn = os.getenv("CIRRUS_BASE_WORKFLOW_ARN") + self.process["workflow"]
+        state_machine_arn, _, execution_name = execution_arn.rpartition(":")
         url = f"s3://{payload_bucket}/{self['id']}/input.json"
 
         # start workflow
-        try:
-            # add input payload to s3
-            s3().upload_json(self, url)
 
-            # create DynamoDB record
-            # -> this overwrites existing states other than PROCESSING
-            wfem.claim_processing(self["id"], payload_url=url)
-
-            # invoke step function
-            self.logger.debug("Running Step Function %s", arn)
-            exe_response = get_client("stepfunctions").start_execution(
-                stateMachineArn=arn,
-                input=json.dumps(self.get_payload()),
-            )
+        # add input payload to s3
+        s3().upload_json(self, url)
```

Comment:
This was previously inside a `try`/`except` block which triggered what appears to be the  functionality now captured in the `_fail_and_raise` function. But it looks like if this step fails for any reason (intermittent error, permissions problem?) we won't have that behavior anymore. Should this get wrapped in its own `try`/`except` with `except` calling `_fail_and_raise`?

Reply:
Yeah, I think you are right: if we can't claim then it seems quite likely we don't want to overwrite the payload in the bucket.